src/components/dashboard/Transactions.vue

```diff
@@ -0,0 +1,173 @@
+<script setup lang="ts">
+import { useConfirm } from "primevue/useconfirm";
+import { useToast } from "primevue/usetoast";
+import {onMounted, ref} from "vue";
+import TransactionDialog from "./TransactionDialog.vue";
+import { useTransactionStore } from '@/stores/transaction';
+
+const confirm = useConfirm();
+const toast = useToast();
+const transaction = useTransactionStore()
+const id = ref<number | undefined>();
+
+const confirmDelete = (id : number) => {
+	confirm.require({
+		message: 'Do you want to delete this record?',
+		header: 'Danger Zone',
+		icon: 'pi pi-info-circle',
+		rejectLabel: 'Cancel',
+		acceptLabel: 'Delete',
+		rejectClass: 'p-button-secondary p-button-outlined',
+		acceptClass: 'p-button-danger',
+		accept: () => {
+			toast.add({ severity: 'error', summary: 'Confirmed', detail: 'Record deleted', life: 3000 });
+			transaction.deleteTransaction(id)
+		},
+	});
+};
+
+onMounted(() => {
+	transaction.setTransactions();
+})
+
+</script>
+
+<template>
+	<Toast />
+	<ConfirmDialog></ConfirmDialog>
+	<TransactionDialog :id='id'/>
+	<div class="dashboard-header">
+		<div class="title">
+		Transaction History
+		</div>
+		<Button label="Add Transaction" icon="pi pi-plus" @click="transaction.toggleVisibility(); id = undefined;" />
+	
+	</div>
+	<div class="card">
+		<DataTable size="large" :value="transaction.transactions" tableStyle="min-width: 50rem" lazy>
+			<Column field="date" header="Date"></Column>
+			<Column field="type" header="Type">
+				<template #body="slotProps">
+					<i  v-if="slotProps.data.type === 'Expenses'" class="pi pi-arrow-up type-icon expense"></i>
+					<i  v-else class="pi pi-arrow-down type-icon income"></i>
+					{{ slotProps.data.type}}
+				</template>
+			</Column>
+			<Column field="category" header="Category"></Column>
+			<Column field="amount">
+				<template #header>
+					<div class="header-title">
+						<i class="pi pi-credit-card" style="margin-right: 5px;"></i>
+						<div class="header-text">Amount</div>
+					</div>
+				</template>
+				<template #body="slotProps">
+					<div class="amount" :style="{ color: slotProps.data.type === 'Expenses' ? '#dc2626' : '#059669' }">
+						{{ slotProps.data.type === "Expenses" ? "-" : "+" }}₱{{ slotProps.data.amount.toFixed(2).replace(/\d(?=(\d{3})+\.)/g, '$&,')}}
```

Comment:
I suggest storing the RegEx code inside a variable for better readability and maintainability.
```suggestion
const whatDoesThisRegExDoVariableName = /\d(?=(\d{3})+\.)/g, '$&,'

						{{ slotProps.data.type === "Expenses" ? "-" : "+" }}₱{{ slotProps.data.amount.toFixed(2).replace(whatDoesThisRegExDoVariableName)}}
```